hs_core/models.py

```diff
@@ -3349,8 +3358,8 @@ def size(self):
         Raises SessionException if iRODS fails.
         """
         # compute the total file size for the resource
-        f_sizes = [f.size for f in self.files.all()]
-        return sum(f_sizes)
+        res_size = self.files.aggregate(Sum('_size'))
```

Comment:
@sblack-usu I am all for db level aggregation for performance reasons, however, in this case, since ```_size``` could be the default ```-1``` until the ```size``` property is called, there may be disconnect there? Perhaps this could be a trigger that we need a management command to populate `_size``` beforehand to make sure res_size is returned correctly?

Reply:
good catch, I'm moving to fast with these comments and not thinking everything through.  Let me think this through. I really like the lazy loading of file size and it will automatically heal if for some reason _size weren't calculated.

Reply:
To keep the lazy loading in place, I guess we can iterate through files to accumulate the file size calling the size property. Have to remove the DB level sum statement.

Reply:
I was thinking of just placing this before the aggregate.
```
[f.size() for f in self.files.filter(_size__gt=0)]
```
Keeps all the logic that can be in the db in the db.

Reply:
Or may be keep the db computation statement as it is. Have the file.size property executed only for those files  that have a negative size value before db sum statement.

Reply:
Yeah, that would work, by triggering size calculation on demand for those files with size less than 0. @sblack-usu I think you can add something like below before the db leval aggregate sum: 
```
for f in self.files.filter(_size__lt=0):
    f.calculate_size()
```

Reply:
@pkdash Are you referring to something different than what I suggested?  (I guess I did have a bug in my suggestion with the comparison).

Reply:
Not much different as per the functionality. But in your suggestion you are creating a list that is not used.

Reply:
right, I didn't use the suggestion as is.  If you're good with the PR as it is now, please approve.